Automattic/jetpack
# JS Components: Convert Spinner component to TypeScript

#### Changes proposed in this Pull Request:
* Convert `Spinner` component to TypeScript and cleans up the propTypes.
* Convert the `Spinner` stories to TS
* Convert the `Spinner` unit tests to TS
* Get rid of `enzyme` in unit test and rather use `@testing-library/react`

Note: Removing enzyme is necessary in order to convert unit tests to TS.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

* Checkout the PR
* Run `cd projects/js-packages/storybook && pnpm run storybook:dev`
* Confirm that all the stories of `Spinner` component work as expected
* Run `jetpack test js-packages/components js`
* Confirm that all the tests pass
* Check the consumers of `Spinner` and related components and confirm that they work as expected
* Check for any regression

<img width="894" alt="Screenshot 2022-05-19 at 3 20 52 PM" src="https://user-images.githubusercontent.com/18226415/169265668-415cb749-5b92-465a-b47d-a53bed353116.png">


---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - 0-as-1202305438007759
